src/game/puppycam2.c

```diff
@@ -478,6 +475,78 @@ void puppycam_reset_values(void) {
     gPuppyCam.splineProgress = 0;
 }
 
+static void create_puppycam1_nodes(void) {
+    u32 i, flagsAdd, flagsRemove;
+    Vec3s pos, diff;
+    struct sPuppyAngles *tempAngle;
+    for (i = 0; i < sizeof(newcam_fixedcam) / sizeof(struct newcam_hardpos); i++) {
+        if (newcam_fixedcam[i].newcam_hard_levelID != gCurrLevelNum || newcam_fixedcam[i].newcam_hard_areaID != gCurrAreaIndex)
+            continue;
+        if ((sPuppyVolumeStack[gPuppyVolumeCount] = mem_pool_alloc(gPuppyMemoryPool, sizeof(struct sPuppyVolume))) == NULL) {
+            gPuppyError |= PUPPY_ERROR_POOL_FULL;
+        #if PUPPYPRINT_DEBUG
+                append_puppyprint_log("Puppycamera volume allocation failed.");
+        #endif
+            return;
+        }
+        if ((tempAngle = mem_pool_alloc(gPuppyMemoryPool, sizeof(struct sPuppyVolume))) == NULL)
+            return;
+        pos[0] = (newcam_fixedcam[i].newcam_hard_X1 + newcam_fixedcam[i].newcam_hard_X2) / 2;
+        pos[1] = (newcam_fixedcam[i].newcam_hard_Y1 + newcam_fixedcam[i].newcam_hard_Y2) / 2;
+        pos[2] = (newcam_fixedcam[i].newcam_hard_Z1 + newcam_fixedcam[i].newcam_hard_Z2) / 2;
+        diff[0] = ABS(newcam_fixedcam[i].newcam_hard_X1 - newcam_fixedcam[i].newcam_hard_X2);
+        diff[1] = ABS(newcam_fixedcam[i].newcam_hard_Y1 - newcam_fixedcam[i].newcam_hard_Y2);
+        diff[2] = ABS(newcam_fixedcam[i].newcam_hard_Z1 - newcam_fixedcam[i].newcam_hard_Z2);
+        vec3s_copy(sPuppyVolumeStack[gPuppyVolumeCount]->pos, pos);
+        vec3s_copy(sPuppyVolumeStack[gPuppyVolumeCount]->radius, diff);
+        sPuppyVolumeStack[gPuppyVolumeCount]->rot = 0; // Doesn't support rotation.
+        sPuppyVolumeStack[gPuppyVolumeCount]->func = newcam_fixedcam[i].newcam_hard_script;
+        tempAngle->yaw = PUPPY_NULL;
```

Comment:
Wat

Reply:
backwards compatability stuff, completely unrelated to current nodes